File: kotlin/llm/src/commonMain/kotlin/dev/deviceai/llm/models/LlmModelInfo.kt
package dev.deviceai.llm.models

/**
 * Describes a GGUF model available for download from HuggingFace.
 *
 * @param id Unique identifier (e.g. "llama-3.2-1b-instruct-q4_k_m")
 * @param name Human-readable name (e.g. "Llama 3.2 1B Instruct Q4_K_M")
 * @param repoId HuggingFace repo (e.g. "bartowski/Llama-3.2-1B-Instruct-GGUF")
 * @param filename GGUF filename in the repo
 * @param sizeBytes Approximate file size in bytes
 * @param quantization Quantization level (e.g. Q4_K_M, Q5_K_M, Q8_0)
 * @param parameters Parameter count label (e.g. "1B", "3B", "7B")
 * @param description Short description of the model
 */
data class LlmModelInfo(
    val id: String,
    val name: String,
    val repoId: String,
    val filename: String,
    val sizeBytes: Long,
    val quantization: String,
    val parameters: String,
    val description: String
)

/**
 * A GGUF model that has been downloaded to local storage.
 *
 * @param id Model identifier matching [LlmModelInfo.id]
 * @param modelPath Absolute path to the .gguf file on disk
 * @param sizeBytes File size in bytes
 */
data class LocalLlmModel(
    val id: String,
    val modelPath: String,
    val sizeBytes: Long
)

File: swift/Sources/DeviceAiLlm/Models/LlmModelInfo.swift
import DeviceAiCore

/// Metadata for a downloadable GGUF model file.
/// Conforms to `ModelInfo` — `ModelManager<LlmModelInfo>` handles download/cache/delete.
public struct LlmModelInfo: ModelInfo {
    public let id: String
    public let displayName: String
    public let sizeBytes: Int64
    public let repoId: String
    public let filename: String
    public let parameters: String
    public let quantization: String
    public let description: String

    public init(id: String, displayName: String, sizeBytes: Int64, repoId: String,
                filename: String, parameters: String, quantization: String, description: String) {
        self.id = id; self.displayName = displayName; self.sizeBytes = sizeBytes
        self.repoId = repoId; self.filename = filename; self.parameters = parameters
        self.quantization = quantization; self.description = description
    }
}
